### PR TITLE
feat: add support for riscv64 architecture

### PR DESCRIPTION
**Pull request by kxxt**
Hi, I would like to add riscv64 support for electron-builder.

I would really appreciate it if you could make a new release shortly after this PR is merged.

### CODE REVIEW

**Comment by kxxt** [comment]
PR updated. The download of electron is skipped for riscv64 architecture because electron doesn't publish binaries for riscv64 now.

BTW it seems that something else is causing the CI to fail.

File: test/src/helpers/downloadElectron.ts [hunk 1]
```diff
@@ -48,6 +48,9 @@ export function downloadAllRequiredElectronVersions(): Promise<any> {
         ? ["ia32", "x64"]
         : require(`${path.join(__dirname, "../../..")}/packages/builder-util/out/util`).getArchCliNames()
     for (const arch of archs) {
+      if (arch === "riscv64")
+        // No prebuilt electron for riscv64
+        continue
```
Comment (mmaietta) [inline test/src/helpers/downloadElectron.ts:53]: Please wrap this with `{}`

```
      if (arch === "riscv64") {
         // No prebuilt electron for riscv64
         continue
      }
```

Comment (kxxt) [reply test/src/helpers/downloadElectron.ts:53]: Done

**Comment by mmaietta** [comment]
Would you be willing to add this test to `linuxPackagerTest.ts`?
```
test.ifNotWindows(
  "AppImage riscv64",
  app({
    targets: Platform.LINUX.createTarget("appimage", Arch.riscv64),
  })
)
```
I'd like to make sure all logic passes with the new arch added

**Comment by kxxt** [comment]
Hi, @mmaietta. Sorry for my late reply.

> I'm not really in favor of fielding github issues that would relate to this

I don't expect there will be such github issues because for now AFAIK the number of developers building electron for riscv64 is very small. And you can ping me for such issues :). 

> If you're looking to provide for a specific arch, I'm wondering if you can use the electronDist function callback directly to pass in your custom electron build?

It almost works. I am packaging wire-desktop for Arch linux riscv64. [It uses electron-builder as a library](https://github.com/wireapp/wire-desktop/blob/linux/3.31.3060/bin/build-tools/lib/build-linux.ts).

The `architecture` becomes `undefined` somehow and caused app-builder to fail. Maybe it is caused by missing riscv64 in the `Arch` enum, which can be fixed by this PR.

**Comment by mmaietta** [comment]
Okay, still not in favor of adding a specific arch that doesn't have an official download supported by electron upstream.

What if we added an `Arch.custom` that requires a `zip` file of the custom electron build? You could therefore use it for any arch and the approach would be reusing the current `electronDist: string | function` in the electron builder config. I think that'd be a more extensible approach.
What are your thoughts?

**Pull request closed without merging**